# Install Package

In [1]:
import pandas as pd
import numpy as np

# Membuat DataFrame

Membuat dataset dengan array dan disimpan dalam variabel dta <br>
Cara 1: Membuat dataframe (Array)

In [4]:
dta=pd.DataFrame(np.array([100,150,50,100,130,120,100,85,70,150,150,120,50,100,100,140,90,150,50,90,120,100,110,75,65]),columns=['Transaction Amount'])
dta

Transaction Amount
0                  100
1                  150
2                   50
3                  100
4                  130
5                  120
6                  100
7                   85
8                   70
9                  150
10                 150
11                 120
12                  50
13                 100
14                 100
15                 140
16                  90
17                 150
18                  50
19                  90
20                 120
21                 100
22                 110
23                  75
24                  65

Cara 2: Membuat dataframe (Dictionary)

In [16]:
dfa=pd.DataFrame({'Transaction Amount':[100,150,50,100,130,120,100,85,70,150,150,120,50,100,100,140,90,150,50,90,120,100,110,75,65]})
dfa

Transaction Amount
0                  100
1                  150
2                   50
3                  100
4                  130
5                  120
6                  100
7                   85
8                   70
9                  150
10                 150
11                 120
12                  50
13                 100
14                 100
15                 140
16                  90
17                 150
18                  50
19                  90
20                 120
21                 100
22                 110
23                  75
24                  65

Cara 3: Membuat dataframe (List)

In [21]:
dla=pd.DataFrame([100,150,50,100,130,120,100,85,70,150,150,120,50,100,100,140,90,150,50,90,120,100,110,75,65],columns=["Transaction Amount"])
dla

Transaction Amount
0                  100
1                  150
2                   50
3                  100
4                  130
5                  120
6                  100
7                   85
8                   70
9                  150
10                 150
11                 120
12                  50
13                 100
14                 100
15                 140
16                  90
17                 150
18                  50
19                  90
20                 120
21                 100
22                 110
23                  75
24                  65

In [25]:
dla.to_excel("Transaction Amount.xlsx", index=False)

# Measure of central tendency dan measure of variability

In [22]:
dta.describe()

Transaction Amount
count           25.000000
mean           102.600000
std             31.823733
min             50.000000
25%             85.000000
50%            100.000000
75%            120.000000
max            150.000000

Berdasarkan output di atas, diperoleh jumlah baris dari data "Transaction Amount" adalah 25, rata rata nya adalah 102.6, nilai minimalnya 50 dan maksimalnya adalah 150 sehingga jangkauannya adalah 100. 

Berdasarkan output di atas juga diperoleh median datanya 100, nilai interkuartilnya 35 karena Q1=85 dan Q3=120, kemudian nilai standar deviasinya 31.823733 sehingga nilai variannya 1012,74998

In [42]:
import statistics
statistics.multimode(dta["Transaction Amount"])

[100]

Berdasarkan output di atas, diperoleh modus dari data "Transaction Amount" adalah 100

In [33]:
print(dta.info())
print("")
print(dta.shape)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25 entries, 0 to 24
Data columns (total 1 columns):
 #   Column              Non-Null Count  Dtype
---  ------              --------------  -----
 0   Transaction Amount  25 non-null     int32
dtypes: int32(1)
memory usage: 232.0 bytes
None

(25, 1)


# Mencari data berdasarkan lokasi data

Misalnya mencari data pada baris ke 13 dan data pada baris ke 15 sampai 18

In [39]:
print(dta.iloc[13,0])
print(dta.loc[15:18,"Transaction Amount"])

100
15    140
16     90
17    150
18     50
Name: Transaction Amount, dtype: int32


# T Test One tailed

## 1. Menentukan Hipotesis 

H0:Rata rata Transaction Amount = 100 <br>
H1:Rata rata Transaction Amoun > 100 <br>
alpha=0.05

In [50]:
pip install scipy

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.3.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


## 2. Menghitung Statistik Uji t

In [51]:
from scipy import stats
stats.ttest_1samp(a=dta, popmean=100)

TtestResult(statistic=array([0.40850016]), pvalue=array([0.68652848]), df=array([24]))

Berdasarkan output di atas, diperoleh nilai p-value=0.68652848 > alpha=0.05 sehingga H0 diterima 

## 3. Mencari t tabel untuk menentukan Kriteria uji

H0 ditolak jika t hitung > t tabel atau p value < alpha(0,05)

In [52]:
stats.t.ppf(1-0.05,24)

1.7108820799094275

## Keputusan dan Kesimpulan

Berdasarkan output sebelumnya dapat disimpulkan bahwa berdasarkan nilai t H0 diterima karena t hitung(0.40850016) < ttabel(1.7108820799094275) atau p-value (0.68652848) > alpha=0.05

Berdasarkan hasil tersebut, dapat disimpulkan bahwa rata rata hasil penjualan pada perusahaan X tidak meningkat atau masih sama dengan 100. Artinya training  penjualan (sales force training) tidak efektif untuk meningkatkan hasil penjualan pada perusahaan X